### PR TITLE
All multi-value attributes can return a single value by using a singu…

### PR DESCRIPTION
**Pull request by tburgin**
…lar wrapper represented by a property with the non-plural name. Added support for multi-value OrgUnits. Added support for a single DNS Name

### CODE REVIEW

File: Source/MOLCertificate.m [hunk 1]
```diff
@@ -320,70 +320,82 @@ - (NSData *)certData {
...
+    return [self x509ValueForLabel:(__bridge NSString *)kSecOIDOrganizationName
+                    fromDictionary:[self x509SubjectName]];
   }];
 }
 
 - (NSString *)orgUnit {
   return [self memoizedSelector:_cmd forBlock:^id{
-      return [self x509ValueForLabel:(__bridge NSString *)kSecOIDOrganizationalUnitName
-                      fromDictionary:[self x509SubjectName]];
+    return ([self orgUnits].count > 0) ? [self orgUnits][0] : nil;
```
Comment (russellhancox) [inline Source/MOLCertificate.m]: The firstObject property on NSArray is intended for this:

``` objc
return [[self orgUnits] firstObject];
```

Returns nil if the array is empty.

File: Source/MOLCertificate.h [hunk 2]
```diff
@@ -87,6 +87,9 @@
 /**  Organizational Unit Name e.g: "Apple Software" */
```
Comment (russellhancox) [inline Source/MOLCertificate.h]: Would be good to update the description that this is now the first OU, if any.

File: Source/MOLCertificate.h [hunk 3]
```diff
@@ -105,6 +108,9 @@
 /**  Issuer organizational unit. */
```
Comment (russellhancox) [inline Source/MOLCertificate.h]: Same here

File: Source/MOLCertificate.h [hunk 4]
```diff
@@ -114,6 +120,9 @@
 /**  NT Principal Name */
 @property(readonly, nonatomic) NSString *ntPrincipalName;
 
+/**  DNS Name / SAN */
```
Comment (russellhancox) [inline Source/MOLCertificate.h]: And here

**Comment by tburgin** [comment]
Cool, I will update when I get a sec.

**Comment by tburgin** [comment]
Updated.

File: Source/MOLCertificate.m [hunk 5]
```diff
@@ -320,70 +320,82 @@ - (NSData *)certData {
...
+    return [self x509ValueForLabel:(__bridge NSString *)kSecOIDOrganizationName
+                    fromDictionary:[self x509IssuerName]];
   }];
 }
 
 - (NSString *)issuerOrgUnit {
   return [self memoizedSelector:_cmd forBlock:^id{
-      return [self x509ValueForLabel:(__bridge NSString *)kSecOIDOrganizationalUnitName
-                      fromDictionary:[self x509IssuerName]];
+    return ([self issuerOrgUnits].count > 0) ? [self issuerOrgUnits][0] : nil;
```
Comment (russellhancox) [inline Source/MOLCertificate.m]: Missed a use of firstObject here, sorry.

Comment (tburgin) [inline Source/MOLCertificate.m]: Whoops, missed that one. I will update in the AM.

**Pull request merged**